### PR TITLE
Allowing encryption and decryption of whole folders

### PR DESCRIPTION
**Pull request by Gen5WO**
* Allowing the use of folders as input
* Removing os.sep and using just regular forward slash
* Updating readme

### CODE REVIEW

File: xcrypt/xcrypt.py [hunk 1]
```diff
@@ -215,14 +215,14 @@ def main():
         sys.exit(1)
 
     if args.output:
-        if args.output[-1] == os.sep:
+        if args.output[-1] == "/":
```
Comment (galaxy4public) [inline xcrypt/xcrypt.py:218]: Is there any particular reason to revert from os.sep to `/`? I think, os.sep is more correct here.

File: xcrypt/xcrypt.py [hunk 2]
```diff
@@ -215,14 +215,14 @@ def main():
...
             print_v('output set to a directory')
             if not os.path.isdir(args.output):
                 print_v(f'creating output directory "{args.output}"')
-                os.makedirs(args.output)
+                os.makedirs(args.output, mode=777)
```
Comment (galaxy4public) [inline xcrypt/xcrypt.py]: I disagree with explicitly forcing world-writable permissions on the directory creation, this for one would not work on my environment where a policy forbids accessing my user any world writeable directories. I would prefer not to have the permissions explicitly defined for mkdir.

File: xcrypt/xcrypt.py [hunk 3]
```diff
@@ -215,14 +215,14 @@ def main():
...
         else:
-            if len(args.file) > 1:
-                print(f'ERROR: cannot process multiple files into one, check that "-o" argument ends with "{os.sep}"')
+            if len(args.file) > 1 or os.path.isdir(args.file[0]):
+                print(f'ERROR: cannot process multiple files into one, check that "-o" argument ends with /')
```
Comment (galaxy4public) [inline xcrypt/xcrypt.py:225]: Ditto re os.sep, what's wrong with it?

**Comment by galaxy4public** [comment]
I like the changes. If you could address my comments, I would gladly merge it.

**Comment by Gen5WO** [comment]
> I like the changes. If you could address my comments, I would gladly merge it.

There might be an issue on GitHub but I can’t see any comment other than this one… :(

**Review by galaxy4public — CHANGES_REQUESTED** [review]
Sorry, I forgot to submit the review, silly me :)

File: xcrypt/xcrypt.py (hunk 1 above)
Comment (Gen5WO) [reply xcrypt/xcrypt.py:218]: When using on windows, the `os.sep` becomes `\`, if you try to pass that in power shell, it complains that you haven’t provided a drive letter or that the drive letter provided isn’t valid. 

Given that forward slash (`/`) works fine for paths on windows and on any Unix compliant system, I think it’s the easiest and most straightforward way.

File: xcrypt/xcrypt.py (hunk 2 above)
Comment (Gen5WO) [reply xcrypt/xcrypt.py]: This actually was a mistake that didn’t need to be pushed 😂 it was done when troubleshooting why the os.sep was giving me all sorts of errors on windows. This should be reverted

File: xcrypt/xcrypt.py (hunk 3 above)
Comment (Gen5WO) [reply xcrypt/xcrypt.py:225]: See answer above, but in summary for strange compatibility reasons (which I know where your main intentions when using it)

File: xcrypt/xcrypt.py (hunk 1 above)
Comment (galaxy4public) [reply xcrypt/xcrypt.py:218]: OK, it is not a big deal anyway for such a small script and a very narrow audience, I guess.  We can live with this. :)

File: xcrypt/xcrypt.py (hunk 3 above)
Comment (galaxy4public) [reply xcrypt/xcrypt.py:225]: OK, we are in agreement here, I believe.

File: xcrypt/xcrypt.py (hunk 2 above)
Comment (galaxy4public) [reply xcrypt/xcrypt.py]:
```suggestion
                os.makedirs(args.output)
```

**Pull request merged**